### PR TITLE
Bump docker images for z3 4.8.10

### PR DESCRIPTION
**Pull request by leonardoalt**
Once the images are built, their hashes will be used by the PR that updates the tests.

### CODE REVIEW

**Review by cameel — DISMISSED** [review]
Looks fine. The archlinux test job is failing because CI is still using the old images with Z3 4.8.9, right?

**Comment by leonardoalt** [comment]
@cameel nop, arch installs directly from `pacman` which has 4.8.10, it fails because the tests weren't updated.
Actually I think I ran this too quickly... the ppa builds must be finished to have an effect on the ubuntu builds, they should fail with the old tests and new z3.

**Comment by leonardoalt** [comment]
Ah no, the images are fine. I need to copy the hashes in the comments here into https://github.com/ethereum/solidity/pull/10845, and the tests should pass there. Only after merging that we should merge this one.

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu2004 [hunk 1]
```diff
@@ -22,7 +22,7 @@
 # (c) 2016-2019 solidity contributors.
 #------------------------------------------------------------------------------
 FROM buildpack-deps:focal AS base
-LABEL version="4"
+LABEL version="5"
```
Comment (bshastry) [inline scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:25]: Does the version need to be revised up? iiuc, there are no z3 related changes in this docker file.

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu2004.clang [hunk 2]
```diff
@@ -22,7 +22,7 @@
 # (c) 2016-2019 solidity contributors.
 #------------------------------------------------------------------------------
 FROM buildpack-deps:focal AS base
-LABEL version="4"
+LABEL version="5"
```
Comment (bshastry) [inline scripts/docker/buildpack-deps/Dockerfile.ubuntu2004.clang:25]: Same as above.

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu2004 (hunk 1 above)
Comment (leonardoalt) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:25]: I think we need to change it to force the image to be rebuilt with the new z3? It installs static z3 via the PPA

Comment (cameel) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu2004:25]: It installs the latest Z3 package from the PPA, which now is 4.8.10. So the newly built image will be different. Bumping version has the nice side-effect of invalidating Docker cache and ensuring that the image will really be rebuilt.

**Review by bshastry — DISMISSED** [review]
LGTM. Failing test should pass in the other PR but better to update that PR with the docker image hash published by the bot in that PR and verify this before merging this PR.

**Review by cameel — APPROVED** [review]
Should be good to merge as soon as tests pass.

**Comment by leonardoalt** [comment]
I'm slightly worried that the docker images are being built again, this shouldn't be the case... if it is, the hashes are gonna be different and we'll have to do the other PR again

**Comment by cameel** [comment]
Ah, right. That's because you rebased it. They will almost certainly be different unless it's using Docker's cache (and I don't think it is because the build machine is not persistent).

I remember that last time @aarlt was updating the images, he put the PR using the hashes on top of the one building the images. Looks like that was a better strategy.

Anyway, the images that were already built should be good so I think all tests will pass anyway, even if you do not take the new hashes.

**Comment by leonardoalt** [comment]
Yea, I cancelled the new builds so we don't have trash lying around, but should be good

**Pull request merged**